packages/x-charts/src/BarChart/BarLabel/BarLabelPlot.tsx

```diff
@@ -0,0 +1,105 @@
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import { to, useTransition } from '@react-spring/web';
+import type { AnimationData, CompletedBarData } from '../types';
+import { BarLabel } from './BarLabel';
+import { BarLabelFunction } from './types';
+
+const leaveStyle = ({ layout, yOrigin, x, width, y, xOrigin, height }: AnimationData) => ({
+  ...(layout === 'vertical'
+    ? {
+        y: yOrigin,
+        x,
+        height: 0,
+        width,
+      }
+    : {
+        y,
+        x: xOrigin,
+        height,
+        width: 0,
+      }),
+});
+
+const enterStyle = ({ x, width, y, height }: AnimationData) => ({
+  y,
+  x,
+  height,
+  width,
+});
+
+type BarLabelPlotProps = {
+  bars: CompletedBarData[];
+  skipAnimation?: boolean;
+  barLabel?: 'value' | BarLabelFunction;
+};
+
+/**
+ * @ignore - internal component.
+ */
+function BarLabelPlot(props: BarLabelPlotProps) {
+  const { bars, skipAnimation, ...other } = props;
```

Comment:
Could it be a public component such that devs can do

```jsx
<ChartsContainer>
  <BarPlot />
  <BarLabelPlot />
</ChartsContainer>
```

Reply:
Ok, I just seen that `BarLabelPlot` is inside `BarPlot`

One advantage of having them separated is that you can be more precise on the DOM order, meaning something can be inserted between the bars and their labels. But not sure it's usefull

Reply:
My point was to use the plot to calculate the necessary logic in a single place, not sure if the naming is right. 

The structure is like this:
```tsx
<ChartsContainer>
  <BarPlot>
    <BarLabelPlot>
      <BarLabel />
    </BarLabelPlot>
  </BarPlot>
</ChartsContainer>
```